### ISSUE TITLE
on login success, users lose their previous context - 3

### ISSUE THREAD

**Issue by paregorios** [isawnyu/isaw.web#392]
Is it possible (using cookies or referrer magic or something) to make it possible for a user who has just successfully logged into get back to the original context page? I can see two desireable scenarios on successful login. Either of them would be ok.

1. After successful login, user ends up on the ```loggedin``` page, but that page has a dynamically generated link to whatever page they were on when they originated the login process.
2. After successful login, user ends up back on the page they were on when they originated the login process, perhaps with a popup or in-page content item congratulating them on the successful login.

**Comment by alecpm** [comment]
I believe this should be working now on staging. I've added support in our fork of `dm.zope.saml2` for a `came_from` request variable used to override the final post-login destination. I've added some js so that when the login link is clicked a `came_from` cookie is set with the url of the originating page. The `dm.zope.saml2` code should expire/unset the cookie if it finds it, so that it doesn't interfere with any post-login logic. Additionally, I had to modify the varnish config to pass along the `came_from` cookie to the Plone instances.
